src/claim_canister/tests/claim_tests.rs

```diff
@@ -0,0 +1,380 @@
+mod setup;
+use candid::{encode_args, decode_one, Nat, Principal};
+use claim_canister::types::{ClaimError, ClaimInfo, ClaimStatus};
+use setup::setup;
+use std::time::Duration;
+
+const TIMELOCK_PLUS_BUFFER: u64 = 24 * 60 * 60 + 60;
+
+#[test]
+fn test_claim_workflow_with_real_pool_canister() {
+    let (pic, claim_canister, pool_canister, owner) = setup();
+
+    let receiver_bytes = [3u8; 29];
+    let receiver = Principal::from_slice(&receiver_bytes);
+    let amount = Nat::from(1000u64);
+    let desc = String::from("Insurance payout for property damage");
+
+    let res = pic
+        .update_call(
+            claim_canister,
+            owner,
+            "add_claim",
+            encode_args((receiver, amount.clone(), pool_canister, desc)).unwrap(),
+        )
+        .expect("Failed to add claim");
+
+    let decoded: Result<Result<u64, ClaimError>, _> = decode_one(&res);
+    let claim_id = decoded
+        .expect("Failed to decode claim_id")
+        .expect("add_claim returned Err");
+
+    let approve_res = pic
+        .update_call(
+            claim_canister,
+            owner,
+            "approve_claim",
+            encode_args((claim_id,)).unwrap(),
+        )
+        .expect("Approval transport failed");
+    let approve_decoded: Result<Result<(), ClaimError>, _> = decode_one(&approve_res);
+    assert!(approve_decoded.is_ok());
+    assert!(approve_decoded.unwrap().is_ok());
+
+    // Verify claim is approved before timelock
+    let claim_info_res = pic
+        .query_call(
+            claim_canister,
+            owner,
+            "get_claim",
+            encode_args((claim_id,)).unwrap(),
+        )
+        .expect("get_claim transport failed");
+    let claim_info: Option<ClaimInfo> = decode_one(&claim_info_res).expect("decode ClaimInfo failed");
+    assert!(claim_info.is_some());
+    let claim_info = claim_info.unwrap();
+    assert_eq!(claim_info.status, ClaimStatus::Approved);
+
+
+    let one_day_plus_grace: Duration = Duration::from_secs(TIMELOCK_PLUS_BUFFER);
```

Comment:
We discussed on meet to use canister if there is right convertion function Duration
There is one 
https://doc.rust-lang.org/std/time/struct.Duration.html#method.from_nanos

Did you check it at all?